plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/plan/PostgreExecutionPlan.java

```diff
@@ -92,8 +92,30 @@ public String getPlanQueryString() {
         if (oldQuery) {
             return "EXPLAIN " + (verbose ? "VERBOSE " : "") + query;
         } else {
-            boolean doAnalyze = CommonUtils.toBoolean(this.configuration.getParameters().get(PostgreQueryPlaner.PARAM_ANALYSE));
-            return "EXPLAIN (FORMAT XML" + (doAnalyze ? ", ANALYSE" : "") + ") " + query;
+            Map<String, Object> parameters = this.configuration.getParameters();
+            StringBuilder explainStat = new StringBuilder(64);
+            explainStat.append("EXPLAIN (FORMAT XML");
+            for (Map.Entry<String, Object> entry : CommonUtils.safeCollection(parameters.entrySet())) {
+                String key = entry.getKey();
+                if (PostgreQueryPlaner.PARAM_COSTS.equals(key)) {
+                    if (!CommonUtils.toBoolean(entry.getValue())) {
+                        explainStat.append(",COSTS FALSE");
+                    }
+                    continue;
+                }
+                if (PostgreQueryPlaner.PARAM_TIMING.equals(key)) {
+                    if (!CommonUtils.toBoolean(entry.getValue())) {
+                        explainStat.append(",TIMING FALSE");
+                    }
+                    continue;
+                }
+                // In other cases we can just add parameter keyword
+                if (CommonUtils.toBoolean(entry.getValue())) {
+                    explainStat.append(",").append(key);
+                }
```

Comment:
IMHO it's can be written easier and more straightforward

```suggestion
                if (PostgreQueryPlaner.PARAM_COSTS.equals(key) && !CommonUtils.toBoolean(entry.getValue())) {
                    explainStat.append(",COSTS FALSE");
                } else if (PostgreQueryPlaner.PARAM_TIMING.equals(key) && !CommonUtils.toBoolean(entry.getValue())) {
                    explainStat.append(",TIMING FALSE");
                } else if (CommonUtils.toBoolean(entry.getValue())) {
                    explainStat.append(",").append(key); // In other cases we can just add parameter keyword
                }
```